src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -1028,7 +1028,11 @@ private static void VerifyModelForOutVarInNotExecutableCode(SemanticModel model,
 
             var typeInfo = model.GetTypeInfo(decl);
             Assert.Equal(expectedType, typeInfo.Type);
-            Assert.Equal(expectedType, typeInfo.ConvertedType);
+            if ((expectedSymbol as LocalSymbol)?.DeclarationKind == LocalDeclarationKind.OutVariable)
```

Comment:
>if ((expectedSymbol as LocalSymbol)?.DeclarationKind == LocalDeclarationKind.OutVariable) [](start = 12, length = 89)

It looks like this method is also called for fields, the condition is not covering them. Also, it is not clear why the assert for ```ConvertedType``` fails for misplaced declaration expressions. It doesn't look like it fails for the ```Type``` property. It feels like we should assert something for scenarios like this (is it null, is it an error type, is it ...) rather than do no validation because ```ConvertedType``` is a public API.